src/schema.rs

```diff
@@ -421,10 +410,102 @@ impl Schema {
         }
     }
 
-    /// Parse a `serde_json::Value` representing a primitive Avro type into a
-    /// `Schema`.
-    fn parse_primitive(primitive: &str) -> AvroResult<Self> {
-        match primitive {
+    /// Create a `Schema` from a string representing a JSON Avro schema.
+    pub fn parse_str(input: &str) -> Result<Schema, Error> {
+        let mut parser = Parser::default();
+        parser.parse_str(input)
+    }
+
+    /// Create a array of `Schema`'s from a list of named JSON Avro schemas (Record, Enum, and
+    /// Fixed).
+    ///
+    /// It is allowed that the schemas have cross-dependencies; these will be resolved
+    /// during parsing.
+    ///
+    /// If two of the input schemas have the same fullname, an Error will be returned.
+    pub fn parse_list(input: &[&str]) -> Result<Vec<Schema>, Error> {
+        let mut input_schemas: HashMap<String, Value> = HashMap::with_capacity(input.len());
+        let mut input_order: Vec<String> = Vec::with_capacity(input.len());
+        for js in input {
+            let schema: Value = serde_json::from_str(js).map_err(Error::ParseSchemaJson)?;
+            if let Value::Object(inner) = &schema {
+                let fullname = Name::parse(&inner)?.fullname(None);
+                let previous_value = input_schemas.insert(fullname.clone(), schema);
+                if previous_value.is_some() {
+                    return Err(Error::NameCollision(fullname));
+                }
+                let _ = input_order.push(fullname);
```

Comment:
nit: this can probably just be `input_order.push(fullname)` 🙂

Reply:
:+1: